### PR TITLE
Separate the "add new config-node" option into a new (+) button

### PR DESCRIPTION
**Pull request by GogoVega**
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Proposed changes

This PR reduces the number of three steps into one to add a config-node by clicking a new (+) button.

The "edit" button will be disabled if no option is available. The "add" option is removed from the list as soon as it contains at least one option.

Requested by @joepavitt on the [forum](https://discourse.nodered.org/t/proposal-config-nodes/86629).

PS: I don't know if I should also adapt the test files? (`10-mqtt_page.js` and `22-websocket_page.js`)

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [x] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `npm run test` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: packages/node_modules/@node-red/editor-client/locales/en-US/editor.json [hunk 1]
```diff
@@ -442,7 +442,7 @@
         "addNewConfig": "Add new __type__ config node",
         "editNode": "Edit __type__ node",
         "editConfig": "Edit __type__ config node",
-        "addNewType": "Add new __type__...",
+        "addNewType": "Please add new __type__...",
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/locales/en-US/editor.json:445]:
```suggestion
        "addNewType": "Add new __type__...",
```

File: packages/node_modules/@node-red/editor-client/locales/fr/editor.json [hunk 2]
```diff
@@ -442,7 +442,7 @@
     "addNewConfig": "Ajouter un nouveau noeud de configuration __type__",
     "editNode": "Modifier le noeud __type__",
     "editConfig": "Modifier le noeud de configuration __type__",
-    "addNewType": "Ajouter un nouveau __type__...",
+    "addNewType": "Veuillez ajouter un nouveau __type__...",
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/locales/fr/editor.json:445]:
```suggestion
    "addNewType": "Ajouter un nouveau __type__...",
```

**Review by knolleary — COMMENTED** [review]
Given the new button takes up horizontal space, making the select box narrower, I don't think we should also change the 'add new' message to push the type further to the right.

Also, by removing the `_ADD_` option from the select box if there are existing config nodes, it is no longer possible to leave the property unset. It forces you to pick an existing config node even if that isn't what you want it to do.

It may be an edge case, but being able to unset the value is something we should allow for.

**Comment by GogoVega** [comment]
Do you want me to add a “nothing” option? So there will be 2 basic options (add and nothing) then the different options plus nothing?

**Comment by GogoVega** [comment]
Something like that?

https://github.com/node-red/node-red/assets/92022724/99883f8a-8104-453d-b9eb-dd7b9bdcd6af

**Comment by knolleary** [comment]
> Something like that?

... maybe... I'm not sure. It's one of the reasons this particular bit of UI has not received the update it has needed for so long. Trying to find the right UI that works for the different valid states it can be in.

I do agree having the `+` button is a good improvement. I'm going to have a play around to see what works.

**Comment by joepavitt** [comment]
Just having a "None" option would work imo

**Comment by GogoVega** [comment]
I am hesitating between the following for the case without options (no node available):
- "add new" (1 option) - the current
- "none" and "add new" (2 options)
- "add or leave blank" (1 option)

Then available options and "none" option.

**Comment by joepavitt** [comment]
"None" _or_ "Add new..." make sense to me. Definitely not both together

File: packages/node_modules/@node-red/editor-client/src/js/ui/editor.js [hunk 3]
```diff
@@ -341,8 +341,9 @@ RED.editor = (function() {
             nodeValue = node[property]
         }
 
-        const buttonId = `${prefix}-lookup-${property}`
-        const selectId = prefix + '-' + property
+        const addBtnId = `${prefix}-add-${property}`;
+        const editBtnId = `${prefix}-lookup-${property}`;
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/editor.js:345]:
```suggestion
        const addBtnId = `${prefix}-btn-${property}-add`;
        const editBtnId = `${prefix}-btn-${property}-edit`;
```

File: packages/node_modules/@node-red/editor-client/src/js/ui/editor.js [hunk 4]
```diff
@@ -365,40 +366,62 @@ RED.editor = (function() {
...
         // create the edit button
-        const btn = $('<a id="' + buttonId + '" class="red-ui-button"><i class="fa fa-pencil"></i></a>')
+        const editButton = $('<a id="' + editBtnId + '" class="red-ui-button"><i class="fa fa-pencil"></i></a>')
             .css({ "margin-left": "10px" })
             .appendTo(outerWrap);
 
+        // create the add button
+        const addButton = $('<a id="' + addBtnId + '-add" class="red-ui-button"><i class="fa fa-plus"></i></a>')
+            .css({ "margin-left": "10px" })
+            .appendTo(outerWrap);
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/editor.js:380]:
```suggestion
        RED.popover.tooltip(editButton, RED._('editor.editConfig', { type }));

        // create the add button
        const addButton = $('<a id="' + addBtnId + '" class="red-ui-button"><i class="fa fa-plus"></i></a>')
            .css({ "margin-left": "10px" })
            .appendTo(outerWrap);
        RED.popover.tooltip(addButton, RED._('editor.addNewConfig', { type }));
```

File: packages/node_modules/@node-red/editor-client/src/js/ui/editor.js [hunk 5]
```diff
@@ -365,40 +366,62 @@ RED.editor = (function() {
...
+            const selectedOpt = select.find(":selected");
+            const optionsLength = select.find("option").length;
             if (selectedOpt?.data('env')) {
-                disableButton(true)
+                disableButton(addButton, true);
+                disableButton(editButton, true);
+            // disable the edit button if no options available
+            } else if (optionsLength === 1 && selectedOpt.val() === "_ADD_") {
+                disableButton(addButton, false);
+                disableButton(editButton, true);
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/editor.js:411]:
```suggestion
                disableButton(editButton, true);
            } else if (selectedOpt.val() === "") {
                disableButton(addButton, false);
                disableButton(editButton, true);
```

File: packages/node_modules/@node-red/editor-client/src/js/ui/editor.js [hunk 6]
```diff
@@ -892,7 +915,10 @@ RED.editor = (function() {
                     }
                 }
 
-                select.append('<option value="_ADD_"'+(value===""?" selected":"")+'>'+RED._("editor.addNewType", {type:label})+'</option>');
+                if (!configNodes.length) {
+                    select.append('<option value="_ADD_" selected>' + RED._("editor.addNewType", { type: label }) + '</option>');
+                }
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/editor.js:920]:
```suggestion
                } else {
                    select.append('<option value="">' + RED._("editor.inputs.none") + '</option>');
                }
```

**Review by knolleary — CHANGES_REQUESTED** [review]
Some minor changes to add a 'none' option, and add tooltips to the buttons

**Review by knolleary — APPROVED** [review]

**Pull request merged**